src/init.cpp

```diff
@@ -1503,6 +1503,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     if (GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION))
         StartTorControl(threadGroup, scheduler);
 
+    SetNodeStartHeight(chainActive.Height());
```

Comment:
Agreed that the name is unclear, will fix.

After some thought, I also agree with using a signal. This will eventually connect to a connman instance, which shouldn't be a problem. I'll update with those suggestions and drop the RFC tag.